### PR TITLE
8346159: Disable CDS AOTClassLinking tests for JVMCI due to JDK-8345635

### PR DESCRIPTION
**Pull request by iklam**
Please review this small change to disable some CDS tests that fail due to JDK-8345635.

Note that I also added `@requires vm.cds.supports.aot.class.linking` to some tests that are using `-XX:+AOTClassLinking` but missed this tag.

Note: the fix for JDK-8345635 (https://github.com/openjdk/jdk/pull/22687) is causing other issues so it's not ready to be integrated yet.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8346159](https://bugs.openjdk.org/browse/JDK-8346159): Disable CDS AOTClassLinking tests for JVMCI due to JDK-8345635 (**Sub-task** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) Review applies to [40ef24be](https://git.openjdk.org/jdk/pull/22730/files/40ef24be65335f7850b857833903b49c8328fb9f)
 * [Doug Simon](https://openjdk.org/census#dnsimon) (@dougxc - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22730/head:pull/22730` \
`$ git checkout pull/22730`

Update a local copy of the PR: \
`$ git checkout pull/22730` \
`$ git pull https://git.openjdk.org/jdk.git pull/22730/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22730`

View PR using the GUI difftool: \
`$ git pr show -t 22730`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22730.diff">https://git.openjdk.org/jdk/pull/22730.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22730#issuecomment-2540542646)
</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Seems reasonable.

Thanks

File: test/hotspot/jtreg/runtime/cds/appcds/LambdaWithUseImplMethodHandle.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
  * @bug 8290417
  * @summary CDS cannot archive lambda proxy with useImplMethodHandle
  * @requires vm.cds
+ * @requires vm.cds.supports.aot.class.linking
+ * @requires !vm.jvmci.enabled
```
Comment (dougxc) [inline test/hotspot/jtreg/runtime/cds/appcds/LambdaWithUseImplMethodHandle.java:31]: Can `@requires` lines have comments? It wouldn't hurt to call out that this is a workaround pending resolution of [JDK-8345969](https://bugs.openjdk.org/browse/JDK-8345969) (or some other issue).

Comment (iklam) [reply test/hotspot/jtreg/runtime/cds/appcds/LambdaWithUseImplMethodHandle.java:32]: I added comments as you suggested.

**Review by dougxc — APPROVED** [review]
Still good.

**Comment by iklam** [comment]
Thanks @dholmes-ora and @dougxc for the review
/integrate

**Pull request closed without merging**